hawkular-alerts-engine/src/main/resources/org/hawkular/alerts/engine/rules/ConditionMatch.drl

```diff
@@ -378,7 +379,37 @@ rule AlertOnSatisfiedDampening
                 notifications.send(notification);
             }
         }
-        insert(  newAlert );
+
+        //insert(  newAlert );
+
         $d.reset();
         update( $d )
+
+        if ($t.isSafetyEnabled()) {
+            if (log != null && log.isDebugEnabled()) {
+                log.debug("Setting Trigger to Safety Mode! " + $t);
+            }
+            $t.setMode(Mode.SAFETY);
+            update( $t )

```

Comment:
This is an interesting point.  I agree that the UI may want to show the current mode although I'm wondering whether Trigger.mode should even be persisted.  It may be best as a transient field, it's not really part of the Trigger definition but rather a runtime value.  If the server is restarted I think it may be OK if all Triggers go back to Firing mode.  We should discuss this behavioral point further and probably get some feedback from others.